src/ProductConstructionService/ProductConstructionService.Api/VirtualMonoRepo/VmrConfiguration.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Maestro.DataProviders;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace ProductConstructionService.Api.VirtualMonoRepo;
+
+public static class VmrConfiguration
+{
+    public const string VmrPathKey = "VmrPath";
+    public const string TmpPathKey = "TmpPath";
+    public const string VmrUriKey = "VmrUri";
+
+    public const string VmrReadyHealthCheckName = "VmrReady";
+    public const string VmrReadyHealthCheckTag = "vmrReady";
+
+    public static void AddVmrRegistrations(this WebApplicationBuilder builder, string vmrPath, string tmpPath, string vmrUri)
+    {
+        builder.Services.TryAddSingleton<IBasicBarClient, SqlBarClient>();
+        builder.Services.AddVmrManagers(
+            "git",
+            vmrPath,
+            tmpPath,
+            builder.Configuration["BotAccount-dotnet-bot-repo-PAT"],
+            builder.Configuration["dn-bot-all-orgs-code-r"]);
+
+        if (!builder.Environment.IsDevelopment())
+        {
+            builder.Services.AddSingleton(new VmrClonerBackgroundServiceOptions(vmrUri));
```

Comment:
So what happens for local development? We need to clone the VMR somewhere ourselves?

Reply:
Yes, if someone is running the service locally, we expect them to have the VMR on their disk already. We can maybe have a flag for letting people check it out?

Reply:
Ah, I see it's in the instructions